docs/guides/nextjs-setup.md

```diff
@@ -0,0 +1,285 @@
+- Would you like to customize the default import alias (@/*)? `No`
+
+### Run the Next.js app
+
+```tsx
+cd react-sdk-walkthrough
+yarn dev
+```
+
+Open a browser to the location output for `local` (e.g., [http://localhost:3000](http://localhost:3000/))
+
+## Eppo flag setup
+1. [Create and copy an SDK key](/sdks/sdk-keys.md) in your Eppo app if you don't already have one.
+2. [Create a feature flag](/quick-starts/feature-flag-quickstart.md/#2-create-a-flag) in your Eppo app.
+3. Make sure your flag is turned on the environment that correlates to your SDK key.
+
+## Client rendered component setup
+
+### Install the Eppo JS client SDK
+
+```bash
+yarn add @eppo/js-client-sdk
+```
+
+### Create a randomization provider
+
+1. Create a provider based off our [Usage in React](/sdks/client-sdks/javascript#usage-in-react) documentation
+2. Create a new `/app` file called `EppoRandomizationProvider.tsx`
+3. Copy and paste the following example code into `EppoRandomizationProvider.tsx`:
+```tsx
+import { ReactElement, useEffect, useState } from "react";
+
+import { init } from "@eppo/js-client-sdk";
+
+interface IEppoRandomizationProvider {
+  waitForInitialization?: boolean;
+  children: ReactElement;
+  loadingComponent?: ReactElement;
+}
+
+const EppoRandomizationProvider = ({
+  waitForInitialization = true,
+  children,
+  loadingComponent = <div>Loading...</div>,
+}: IEppoRandomizationProvider): ReactElement => {
+  const [isInitialized, setIsInitialized] = useState(false);
+  useEffect(() => {
+    init({
+      apiKey: "<SDK-KEY>",
+      assignmentLogger: {
+        logAssignment(assignment) {
+          console.log('TODO: log', assignment)
+        },
+      },
+    }).then(() => {
+      return setIsInitialized(true);
+    });
+  }, []);
+
+  if (!waitForInitialization || isInitialized) {
+    return children;
+  }
+  return loadingComponent;
+}
+
+export default EppoRandomizationProvider;
+```
+4. Replace API key placeholder with your SDK key
+
+### Create a component to use the randomization provider
+1. Create a new `/app` file  `OfferExperiment.tsx`
+2. Copy and paste the following code into the `OfferExperiment.tsx` file.
+```tsx
+import { useMemo } from "react";
+import { getInstance } from "@eppo/js-client-sdk";
+
+const OfferComponent = () => {
+
+    const subjectKey = "<SUBJECT-KEY>"
+
+    const assignedVariation = useMemo(() => {
+      const eppoClient = getInstance();
+      return eppoClient.getStringAssignment("<FLAG-KEY>", subjectKey, <SUBJECT-ATTRIBUTES>, "<DEFAULT-VALUE>");
+    }, []);
+  
+    return (
+      <div>
+        {assignedVariation === "control" && <p>50% off widgets!</p>}
+        {assignedVariation === "test" && <p>Buy one widget get one free!</p>}
+        {!assignedVariation && <p>Buy a widget today</p>}
+      </div>
+    );
+  }
+
+export default OfferComponent;
+```
+In the example above the disabled/error (`null`) case is handled as well as assigned variations. 
```

Comment:
Should never be `null` right? Since this is using v3 of the SDK

Reply:
ah good catch, this was written in the before times